src/devices/custom_devices_diy.ts

```diff
@@ -148,6 +153,22 @@ const tzLocal = {
             };
         },
     } as Tz.Converter,
+    multi_zig_sw_switch_type: {
+        key: ['switch_type'],
+        convertGet: async (entity, key, meta) => {
+            await entity.read('genOnOffSwitchCfg', ['switchType']);
+        },
+        convertSet: async (entity, key, value, meta) => {
+            if (key == 'switch_type') {
+                const data = getListValueByKey(switchTypesList, value as string);
```

Comment:
Sorry for not noticing earlier, but why not use `utils.getFromLookup(value, switchTypesList);` here?